setup_test.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for setup.py."""
+
+from __future__ import annotations
+
+import builtins
+import os
+import sys
+
+from core.tests import test_utils
+
+import pkg_resources
+import setuptools
+
+from scripts import common # isort:skip pylint: disable=unused-import
+# Since feconf imports typing_extensions, it should be
+# imported after common is imported.
+from core import feconf # isort:skip
+
+
+class SetupTests(test_utils.GenericTestBase):
+    """Unit tests for setup.py."""
+
+    def test_setuptools_is_invoked_with_correct_parameters(self) -> None:
+        packages = (
+            'module1==2.1.2\n'
+            'module2==3.2.3\n'
+            'module3==4.3.4\n')
+        with open('dummy_requirements.txt', 'w', encoding='utf-8') as f:
+            f.write(packages)
+
+        dummy_file_object = open('dummy_requirements.txt', encoding='utf-8')
+
+        swap_open = self.swap_with_checks(
+            builtins, 'open',
+            lambda *unused_args, **unused_kwargs: dummy_file_object,
+            expected_args=(('requirements.txt',),))
+
+        required_packages = [
+            str(requirement)
+            for requirement in pkg_resources.parse_requirements(packages)
+        ]
```

Comment:
This seems unnecessary since `required_packages` is just going to be `['module1==2.1.2', ...]`:

```python
>>> import pkg_resources
>>> pkg_resources.parse_requirements('module1==2.1.2')
<generator object parse_requirements at 0x10fe078d0>
>>> [str(res) for res in pkg_resources.parse_requirements('module1==2.1.2')]
['module1==2.1.2']
```

Couldn't you just write something like this:

```python
required_packages = [
    'module1==2.1.2',
    'module2==3.2.3',
    'module3==4.3.4',
]
```

Reply:
Updated it now. Thanks!